ts/components/session/conversation/SessionConversation.tsx

```diff
@@ -0,0 +1,860 @@
+// tslint:disable: no-backbone-get-set-outside-model
+
+import React from 'react';
+import classNames from 'classnames';
+
+import { ConversationHeader } from '../../conversation/ConversationHeader';
+import { SessionCompositionBox } from './SessionCompositionBox';
+import { SessionProgress } from '../SessionProgress'
+
+import { Message } from '../../conversation/Message';
+import { FriendRequest } from '../../conversation/FriendRequest';
+import { TimerNotification } from '../../conversation/TimerNotification';
+
+import { getTimestamp } from './SessionConversationManager';
+
+import { SessionScrollButton } from '../SessionScrollButton';
+import { SessionGroupSettings } from './SessionGroupSettings';
+import { ResetSessionNotification } from '../../conversation/ResetSessionNotification';
+
+
+interface State {
+  conversationKey: string;
+  sendingProgress: number;
+  prevSendingProgress: number;
+  // Sending failed:  -1
+  // Not send yet:     0
+  // Sending message:  1
+  // Sending success:  2
+  sendingProgressStatus: -1 | 0 | 1 | 2;
+
+  unreadCount: number;
+  messages: Array<any>;
+  selectedMessages: Array<string>;
+  isScrolledToBottom: boolean;
+  doneInitialScroll: boolean;
+  displayScrollToBottomButton: boolean;
+  messageFetchTimestamp: number;
+
+  showRecordingView: boolean;
+  showOptionsPane: boolean;
+  showScrollButton: boolean;
+}
+
+export class SessionConversation extends React.Component<any, State> {
+  private readonly messagesEndRef: React.RefObject<HTMLDivElement>;
+  private readonly messageContainerRef: React.RefObject<HTMLDivElement>;
+
+  constructor(props: any) {
+    super(props);
+
+    const conversationKey = this.props.conversations.selectedConversation;
+    const conversation = this.props.conversations.conversationLookup[conversationKey];
+    const unreadCount = conversation.unreadCount;
+
+    console.log(`[conv] Conversation:`, conversation);
+
+    this.state = {
+      sendingProgress: 0,
+      prevSendingProgress: 0,
+      sendingProgressStatus: 0,
+      conversationKey,
+      unreadCount,
+      messages: [],
+      selectedMessages: [],
+      isScrolledToBottom: !unreadCount,
+      doneInitialScroll: false,
+      displayScrollToBottomButton: false,
+      messageFetchTimestamp: 0,
+
+      showRecordingView: false,
+      showOptionsPane: false,
+      showScrollButton: false,
+    };
+
+    this.handleScroll = this.handleScroll.bind(this);
```

Comment:
instead of having a ton of `<function>.bind(this)`, you should be able to convert the function to an arrow function and it should work.